### PR TITLE
Php7 shop classes

### PR DESCRIPTION
**Pull request by raiwa**
Shop classe Checklist:

- acttion recorder.php
checked with: contact us message OK

- alertbox.php
checked with: contact us message, create account OK

- boxes.php
checked with: store incl. boxes OK

- breadcrumb.php
checked with: store incl. breadcrumb OK

- category_tree.php
checked with: store incl. category box, category-subcategory links OK

- cc_validation.php
checked with: checkout psigate cc payment OK

- currencies.php
checked with: general store pages with price, checkout OK

- e-mail.php
checked with: contact us message, create account message, order confirmation message OK

- http_client.php
checked with: checkout_shipping OK

- language.php
checked with: application_top.php, multilanguage store language changed OK

- message_stack.php:
checked with: contact us, create account OK

- mime.php:
checked with: contact us, Admin "Use MIME HTML When Sending Emails" set to "true" OK

- navigation_history.php:
checked with: guest => add to cart => checkout => login => checkout_shipping OK

- order.php:
checked with: complete checkout OK

- order_total.php:
checked with: complete checkout OK

- osc_template.php:
checked with: general store layout, modules OK

- passwordhash.php:
checked with: password forgotten OK

- payment.php:
checked with: complete checkout OK

- shipping.php:
checked with: complete checkout OK

- shopping_cart.php:
checked with: add to cart, cart update qty., cart remove, complete checkout OK

- split_page_result.php:
checked with: product_listing, new products OK

### CODE REVIEW

**Comment by BrockleyJohn** [comment]
@raiwa Hi Rainer, I've begun testing on this but I can't find any way of testing the changes to boxes.php - I don't think these classes are used any more. I have gone so far as to comment out the include in application_top and it hasn't made any difference... yet.

Am I missing something?

**Comment by raiwa** [comment]
@BrockleyJohn 
You are right. It was used in 2.3.4 standard in message stack class. Now this is replaced by alertBlock.

We might suggest to remove it or may it bee still needed for compatibility with old add-ons? I don't think so.
Is this something for a separate pull request?

rgds
Rainer

**Comment by BrockleyJohn** [comment]
@gburton Gary, I think we need a steer here. The issue is all the old formatting classes in boxes.php.

They are still included in application_top and throw deprecated warnings, but are not used any more in core code (may be used by addons?). Options:
- remove include in application_top [may cause some customised stores to fail straightaway but easy for them to fix]
- fix but don't test [!]
- fix and write some code specially to instantiate classes for testing [may be completely unnecessary]

tableBox
- extensions use as a method
- never instantiated solo - test with extension
- add empty constructor if necessary

infoBox
- extends tableBox
- not used can't test

infoBoxHeading
- extends tableBox
- not used can't test

contentBox
- extends tableBox
- not used can't test

contentBoxHeading
- extends tableBox
- not used can't test

errorBox
- extends tableBox
- not used can't test

productListingBox
- extends tableBox
- not used can't test

**Comment by raiwa** [comment]
@BrockleyJohn,

I checked the boxes.php class with an old osc 2.2rc2a info box.
- copied box "information.php" to catalog/includes/boxes/information.php
- added language definitions to english.php
- manual box call at the end of template_top.php:
  <?php require('includes/boxes/information.php'); ?>

Results:
1. if I comment out in application_top.php:
// infobox
//  require(DIR_WS_CLASSES . 'boxes.php');

Error:
Fatal error: Uncaught Error: Class 'infoBoxHeading' not found in C:\xampp_php_7_0\htdocs\2.3.4BS_Master_2016_04\includes\boxes\information.php:20 Stack trace: #0 C:\xampp_php_7_0\htdocs\2.3.4BS_Master_2016_04\includes\template_top.php(61): require() #1 C:\xampp_php_7_0\htdocs\2.3.4BS_Master_2016_04\index.php(35): require('C:\\xampp_php_7_...') #2 {main} thrown in C:\xampp_php_7_0\htdocs\2.3.4BS_Master_2016_04\includes\boxes\information.php on line 20

Conclusion: infoBoxHeading is used in the old 2.2rc2a information box. Info box content is also used, but further errors are not displayed.

2. If the old box.php class is used, I get these deprecated errors:
Deprecated: Methods with the same name as their class will not be constructors in a future version of PHP; tableBox has a deprecated constructor in C:\xampp_php_7_0\htdocs\2.3.4BS_Master_2016_04\includes\classes\boxes.php on line 13

Deprecated: Methods with the same name as their class will not be constructors in a future version of PHP; infoBox has a deprecated constructor in C:\xampp_php_7_0\htdocs\2.3.4BS_Master_2016_04\includes\classes\boxes.php on line 75

Deprecated: Methods with the same name as their class will not be constructors in a future version of PHP; infoBoxHeading has a deprecated constructor in C:\xampp_php_7_0\htdocs\2.3.4BS_Master_2016_04\includes\classes\boxes.php on line 100

Deprecated: Methods with the same name as their class will not be constructors in a future version of PHP; contentBox has a deprecated constructor in C:\xampp_php_7_0\htdocs\2.3.4BS_Master_2016_04\includes\classes\boxes.php on line 132

Deprecated: Methods with the same name as their class will not be constructors in a future version of PHP; contentBoxHeading has a deprecated constructor in C:\xampp_php_7_0\htdocs\2.3.4BS_Master_2016_04\includes\classes\boxes.php on line 148

Deprecated: Methods with the same name as their class will not be constructors in a future version of PHP; errorBox has a deprecated constructor in C:\xampp_php_7_0\htdocs\2.3.4BS_Master_2016_04\includes\classes\boxes.php on line 165

Deprecated: Methods with the same name as their class will not be constructors in a future version of PHP; productListingBox has a deprecated constructor in C:\xampp_php_7_0\htdocs\2.3.4BS_Master_2016_04\includes\classes\boxes.php on line 172

If I use the updated box.php class all shows correct without errors.


General conclusion:
The update fixes are correct.
However the question is if we need still to keep the box.php class to offer compatibility with osc 2.2 boxes.

**Comment by BrockleyJohn** [comment]
@raiwa Yes, I walked the code and it desk-checks fine ;)

There's a bigger problem though, your commits have too much change in them - all the lines of each file are deleted and reinserted, not just where the changes are. This includes files that haven't been changed at all (because they didn't need it) like category_tree.php and cc_validation.php

Do you have any idea why this is?

**Comment by raiwa** [comment]
I'll have a look and resubmit. My GitHub setup seems to be corrupted since I hadn't touch it since a year or so, sorry for this.

**Comment by raiwa** [comment]
@BrockleyJohn,

Could fix it. Now only real changes show.
Not so dificult once you got it, until you get it, hell.

**Comment by BrockleyJohn** [comment]
@raiwa Thanks the changes are better now but there are a couple of extra things: a mod to product_listing module and the whole of order.php. Nearly there!

**Comment by raiwa** [comment]
@BrockleyJohn , it's fixed and I believe it's all ok now. Sorry, still bloody GitHub beginner.

**Comment by BrockleyJohn** [comment]
@raiwa Thanks Rainer, that's great now. I'll finish testing and let you know... I've a feeling the usps bit might take a while to set up - maybe I'll try and elicit some help for that one

**Comment by BrockleyJohn** [comment]
@raiwa - Rainer, what happened?

**Comment by raiwa** [comment]
@BrockleyJohn sorry, tried to clean up my fork and messed it up by accident

**Comment by BrockleyJohn** [comment]
OK - I'm done with testing. All the classes have stopped producing the deprecated warnings and I have validated that all instantiate properly with the exceptions detailed above (all those in boxes.php) plus the class httpClient in http_client.php
The latter requires a properly set-up usps shipping module and is only instantiated when making a request for a quote after all validation is passed. I'm not in a position to make this happen.

**Comment by Kymation** [comment]
The httpClient class is obsolete in that it does not support HTTP 1.1. This makes it fail when used with the older USPS modules, as USPS now uses chunked encoding. 

Do we want to replace this class? There is a newer class available that does support HTTP 1.1, but the modules that depend on it would have to be rewritten to work with it. Or do we just want to drop it and force everyone to use cURL?

**Comment by BrockleyJohn** [comment]
@Kymation - Jim, I infer that the usps shipping module included in core no longer works; is that what you're saying? If so, we should deal with that as a separate issue from the php7 stuff.

**Comment by Kymation** [comment]
The USPS module in core does not work properly and therefore should not be used for testing anything. That is a separate issue except for the effect on testing.

The httpClient class is obsolete and needs to be updated, replaced, or removed. There is no need to do the PHP 7 upgrade as the result will still not work properly.

**Comment by BrockleyJohn** [comment]
Thanks, Jim. What I thought - goes on the "not my testing problem" pile, along with a recommendation to remove the include from checkout_shipping.php

**Comment by gburton** [comment]
@raiwa @BrockleyJohn what is the status of the changes?

**Comment by BrockleyJohn** [comment]
The nearest thing to an overview is the issue conversation. To summarise:

admin side: what needs doing is done (2 pull requests). I have tested them and I think Rainer has too.

shop side: Rainer did classes & I have tested it. I'm about halfway through modules/payment... should have it pull-request ready by the end of the weekend if not sooner
The rest is still to do, but on the whole more readily testable.

What would be nice is some proper regression testing (ie. checking we've not broken anything) in a regular 5.x store. 

Outstanding issues so far:
- basically all variations on module/class no longer used in core code (or not usable), so can't be tested if changed (I have a list of exactly which ones)

**Comment by gburton** [comment]
Thanks @BrockleyJohn 
No rush on it at all, so don't worry about that.

**Comment by jwisdom** [comment]
@raiwa 
On your alertbox change, from what I understand and have tested, you can leave your constuctor as 
`    function __construct($contents, $alert_output = false) {`
Messagebox won't call that constructor unless
`parent::__construct();`
is called.
See php manual
http://tz1.php.net/manual/en/language.oop5.decon.php#object.construct
and stackoverflow answer with examples.
http://stackoverflow.com/a/3477551
